resources/directconnect_gateways.go

```diff
@@ -62,6 +64,124 @@ func DirectconnectGateways() *schema.Table {
+				Resolver:    fetchDirectconnectGatewayAssociations,
+				Columns: []schema.Column{
+					{
+						Name:        "directconnect_gateway_id",
+						Description: "Unique ID of aws_directconnect_gateways table (FK)",
+						Type:        schema.TypeUUID,
+						Resolver:    schema.ParentIdResolver,
+					},
+					{
+						Name:        "allowed_prefixes_to_direct_connect_gateway",
+						Description: "The Amazon VPC prefixes to advertise to the Direct Connect gateway.",
+						Type:        schema.TypeStringArray,
+						Resolver:    fetchDirectconnectGatewayAssociationAllowedPrefixes,
+					},
+					{
+						Name:        "associated_gateway_id",
+						Description: "The ID of the associated gateway.",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("AssociatedGateway.Id"),
+					},
+					{
+						Name:        "associated_gateway_owner_account",
+						Description: "The ID of the AWS account that owns the associated virtual private gateway or transit gateway.",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("AssociatedGateway.OwnerAccount"),
+					},
+					{
+						Name:        "associated_gateway_region",
+						Description: "The Region where the associated gateway is located.",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("AssociatedGateway.Region"),
+					},
+					{
+						Name:        "associated_gateway_type",
+						Description: "The type of associated gateway.",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("AssociatedGateway.Type"),
+					},
+					{
+						Name:        "association_id",
+						Description: "The ID of the Direct Connect gateway association",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "association_state",
+						Description: "The state of the association.",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "direct_connect_gateway_owner_account",
+						Description: "The ID of the AWS account that owns the associated gateway.",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "state_change_error",
+						Description: "The error message if the state of an object failed to advance.",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "virtual_gateway_id",
+						Description: "The ID of the virtual private gateway. Applies only to private virtual interfaces.",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "virtual_gateway_owner_account",
+						Description: "The ID of the AWS account that owns the virtual private gateway.",
+						Type:        schema.TypeString,
+					},
+				},
+			},
+			{
+				Name:        "aws_directconnect_gateway_attachments",
+				Description: "Information about the attachment between a Direct Connect gateway and virtual interfaces.",
+				Resolver:    fetchDirectconnectGatewayAttachments,
+				Columns: []schema.Column{
```

Comment:
I would also save the original DirectConnectGatewayId from AWS as resource_id (hopefully soon we will update the sdk to allow users to define any column as the PK)